api/src/wfl/once.clj

```diff
@@ -52,3 +52,24 @@
     (if (util/getenv "WFL_DEPLOY_ENVIRONMENT")
       (service-account-token)
       (util/shell! "gcloud" "auth" "print-access-token"))))
+
+(defn- read-oauth-client-id
+  "Actively read the OAuth client ID from vault or the environment"
+  []
+  (if-let [id-from-env
+           (not-empty (util/getenv "WFL_OAUTH2_CLIENT_ID"))]
+    id-from-env
+    (if-let [id-from-vault
+             (util/do-or-nil-silently
+               (-> "WFL_DEPLOY_ENVIRONMENT"
+                   (util/getenv "debug")
+                   wfl/error-or-environment-keyword
+                   env/stuff :server :vault
+                   util/vault-secrets :oauth2_client_id))]
+      id-from-vault
+      ;; Client ID for gotc-dev, the old hardcoded value for backwards-compatibility
+      "450819267403-n17keaafi8u1udtopauapv0ntjklmgrs.apps.googleusercontent.com")))
+
+(def return-oauth-client-id
+  "Memoize the reading of the OAuth client ID to be more responsive"
+  (memoize read-oauth-client-id))
```

Comment:
I was wondering when to use `memoize` vs `delay`. This link seems interesting:
https://stackoverflow.com/questions/60760896/clojure-when-to-use-memoize-and-when-to-use-delay-force

Out of interest, what was your reasoning for one over the other?

Reply:
The reasoning is that I didn't know delay existed lol

Reply:
Let the JVM memoize for you. Try memoize if that fails.

Reply:
https://github.com/tbl3rd/clj-intro/blob/05508471656578630dd3f9b00671dbc4e2dfe527/src/017-deref.clj#L213-L239